bindings/python/setup.py

```diff
@@ -187,24 +203,38 @@ def build_extension_with_b2(self):
         if self.cxxstd:
             args.append(f"cxxstd={self.cxxstd}")
 
-        # Jamfile hack to copy the module to our target directory
+        # Jamfile hacks to ensure we select the python environment defined in
+        # our project-config.jam
+        args.append("libtorrent-python=on")
+        args.append(f"python={sysconfig.get_python_version()}")
+
+        # Our goal is to produce an artifact at this path. If we do this, the
+        # distutils build system will skip trying to build it.
         target = pathlib.Path(self.get_ext_fullpath("libtorrent"))
+        self.announce(f"target: {target}")
+
+        # b2 doesn't provide a way to signal the name or paths of its outputs.
+        # We try to convince python.jam to name its output file like our target
+        # and copy it to our target directory.
```

Comment:
it seems like this comment would be more appropriate at the place where we produce the `using` statement in the temporary project config file. Since that's where we actually produce the filename for the module.

Reply:
It's relevant to the next lines: this is why we invoke the `install_module` rule and set the target directory.

It _would_ be relevant to the next part too, invoking `write_b2_python_config()`, if I'd done what you'd done earlier to pass the target filename and de-mangle it to find the right suffix to configure in boost.

But `write_b2_python_config()` can't force b2 to use an arbitrary filename:
 * `Jamfile` configures it to be `libtorrent.*`
 * `python.jam` configures it to be `*.{so,pyd,dll,sl}`
 * Debian actually patches `python.jam` to always enforce it to be exactly the name expected by distutils

So the best guarantee we can offer is that `write_b2_python_config()` ensures b2 will output a name expected by distutils, on all systems.

Moreover, I found that if we pass a target filename to `write_b2_python_config()` and de-mangle it to extract a suffix, it seems like more lines of code, more brittle and less clear than just grabbing the right `EXT_SUFFIX` sysconfig value.

So ... there's no reason to pass the target filename to `write_b2_python_config()`.

I updated some comments to clarify this.